qadence/ml_tools/train_grad.py

```diff
@@ -125,6 +126,22 @@ def loss_fn(model: torch.nn.Module, data: torch.Tensor) -> tuple[torch.Tensor, d
     # initialize tensorboard
     writer = SummaryWriter(config.folder, purge_step=init_iter)
 
+    perform_val = isinstance(config.val_every, int)
+    if perform_val and not isinstance(dataloader, DictDataLoader):
```

Comment:
should we move the first `if` into the second?

Reply:
thanks, done!